### PR TITLE
Preheat: Fix in preheat and filament menu

### PR DESCRIPTION
**Pull request by espr14**

### CODE REVIEW

File: src/gui/screen_menu_filament.cpp [hunk 1]
```diff
@@ -14,7 +14,12 @@
 
 /// Sets temperature of nozzle not to ooze before print (MBL)
 void setPreheatTemp() {
-    marlin_gcode_printf("M104 S%d D%d", (int)PREHEAT_TEMP, (int)filaments[get_filament()].nozzle);
+    const marlin_vars_t *vars = marlin_vars();
+    if (!vars)
```
Comment (RadekVana) [inline src/gui/screen_menu_filament.cpp:18]: no need to check ` if (!vars)`, it cannot be nullptr

File: src/gui/screen_menu_preheat.cpp [hunk 2]
```diff
@@ -14,9 +14,14 @@ class MI_Filament : public WI_LABEL_t {
     virtual void click(Iwindow_menu_t & /*window_menu*/) override {
         const filament_t filament = filaments[T];
         marlin_gcode("M86 S1800"); // enable safety timer
-        marlin_gcode_printf("M104 S%d", (int)filament.nozzle);
+        /// don't use preheat temp for cooldown
+        if (PREHEAT_TEMP >= filament.nozzle) {
```
Comment (RadekVana) [inline src/gui/screen_menu_preheat.cpp:18]: use `if (PREHEAT_TEMP > filament.nozzle)` parameters S and D makes no sense when they are equal

File: src/gui/screen_menu_filament.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/screen_menu_filament.cpp:18]: also missing else

File: src/common/client_response.hpp [hunk 3]
```diff
@@ -48,7 +48,7 @@ constexpr T GetEnumFromPhaseIndex(size_t index) {
 //and YES phase can have 0 responses
 //every enum must have "_first" and "_last"
 //"_first" ==  "previous_enum::_last" + 1
-//EVERY response shall have unique ID (so every button in GIU is unique)
+//EVERY response shall have unique ID (so every button in GUI is unique)
```
Comment (DRracer) [inline src/common/client_response.hpp:51]: when you are fixing the comment wording, please make sure it is proper English too ;) 
... shall have **a** unique ID

File: src/gui/screen_menu_filament.cpp [hunk 4]
```diff
@@ -14,7 +14,12 @@
...
+        marlin_gcode_printf("M104 S0");
+
+    /// don't read from EEPROM since it's not in sync
```
Comment (DRracer) [inline src/gui/screen_menu_filament.cpp:21]: EEPROM?

Comment (espr14) [reply src/gui/screen_menu_filament.cpp:21]: Previous implementation that does not work.

File: src/gui/screen_menu_preheat.cpp (hunk 2 above)
Comment (espr14) [reply src/gui/screen_menu_preheat.cpp:18]: In case of equality "M104 S%d" is called.

Comment (espr14) [reply src/gui/screen_menu_preheat.cpp:18]: S makes sense in both cases.

**Pull request merged**